### PR TITLE
Update `react-native-maps` to 1.3.2

### PR DESCRIPTION
**Pull request by aleqsio**
# Why
Update for sdk 47
Closes ENG-6534.

# How

There were a lot of native changes and a major version bump.
The following manual change needed to be made to `ios/Podfile`:
```diff
- pod 'GoogleMaps', '~> 3.6'
+ pod 'GoogleMaps', '~> 7.1'
- pod 'Google-Maps-iOS-Utils', '~> 2.1.0'
+ pod 'Google-Maps-iOS-Utils', '~> 4.1.0'
  ```
In addition, Expo home uses `react-native-maps` so fixing a breaking change in typescript types was also required.

# Test Plan

Tested manually in Expo Go Android and iOS as well in Bare Expo.

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,11 +13,13 @@ Package-specific changes not released in any SDK will be added here just before
 - Updated `react-native-view-shot` from `3.3.0` to `3.4.0`. ([#19405](https://github.com/expo/expo/pull/19405) by [@douglowder](https://github.com/douglowder))
 - Updated `react-native-webview` from `11.23.0` to `11.23.1`. ([#19375](https://github.com/expo/expo/pull/19375) by [@aleqsio](https://github.com/aleqsio))
 - Updated `react-native-gesture-handler` from `2.5.0` to `2.7.0`. ([#19362](https://github.com/expo/expo/pull/19362) by [@tsapeta](https://github.com/tsapeta))
+<<<<<<< HEAD
```
Comment (tsapeta) [inline CHANGELOG.md]:
```suggestion
```
😅

**Review by tsapeta — COMMENTED** [review]
I think fabric-tester submodule update needs to be reverted 🤔

File: CHANGELOG.md (hunk 1 above)
Comment (aleqsio) [reply CHANGELOG.md]: Yeah, saw that too, but you were just a sec too quick 😢

File: CHANGELOG.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Package-specific changes not released in any SDK will be added here just before
 - Updated `react-native-screens` from `3.15.0` to `3.18.0`. ([#19383](https://github.com/expo/expo/pull/19383) by [@tsapeta](https://github.com/tsapeta))
 - Updated `@shopify/react-native-skia` from `0.1.136` to `0.1.153`. ([#19360](https://github.com/expo/expo/pull/19360) by [@kudo](https://github.com/kudo))
 - Updated `@react-native-community/datetimepicker` from `6.2.0` to `6.5.0`. ([#19419](https://github.com/expo/expo/pull/19419) by [@byCedric](https://github.com/byCedric))
+- Updated `react-native-maps` from `0.31.1` to `1.3.2`.
```
Comment (tsapeta) [inline CHANGELOG.md:21]:
```suggestion
- Updated `react-native-maps` from `0.31.1` to `1.3.2`. ([#19414](https://github.com/expo/expo/pull/19414) by [@aleqsio](https://github.com/aleqsio))
```

**Review by tsapeta — APPROVED** [review]

**Comment by aleqsio** [comment]
Merging as it seems the issue is the same as caused by:
https://github.com/expo/expo/pull/19419

**Pull request merged**